collector/src/compile/benchmark/mod.rs

```diff
@@ -620,13 +620,23 @@ mod tests {
     #[test]
     fn check_compile_benchmarks() {
         // Check that we can deserialize all perf-config.json files in the compile benchmark
-        // directory.
+        // directory and that they have [workspace] in their Cargo.toml.
         let root = env!("CARGO_MANIFEST_DIR");
-        let benchmarks = get_compile_benchmarks(
-            &Path::new(root).join("compile-benchmarks"),
-            CompileBenchmarkFilter::All,
-        )
-        .unwrap();
+        let benchmark_dir = Path::new(root).join("compile-benchmarks");
+        let benchmarks =
+            get_compile_benchmarks(&benchmark_dir, CompileBenchmarkFilter::All).unwrap();
         assert!(!benchmarks.is_empty());
+
+        for benchmark in benchmarks {
+            let dir = benchmark_dir.join(&benchmark.name.0);
+            let cargo_toml = std::fs::read_to_string(&dir.join("Cargo.toml"))
+                .expect(&format!("Cannot read Cargo.toml of {}", benchmark.name));
+            if !cargo_toml.contains("[workspace]") {
```

Comment:
Maybe as an assert

Reply:
Had that previously, but it didn't contain the right error message. I forgot that we can add arbitrary formatted messages to asserts :facepalm: Changed.